File: sparkcql-fhir/src/main/scala/gov/va/sparkcql/fhir/FhirModelAdapterFactory.scala
package gov.va.sparkcql.fhir

import gov.va.sparkcql.adapter.model.{ModelAdapter, ModelAdapterFactory}
import gov.va.sparkcql.di.ComponentConfiguration

class FhirModelAdapterFactory extends ModelAdapterFactory {

  override def create(configuration: ComponentConfiguration): ModelAdapter = new FhirModelAdapter()
}

File: sparkcql-fhir/src/main/java/gov/va/sparkcql/fhir/FhirModelAdapterFactory.java
package gov.va.sparkcql.fhir;

import org.apache.spark.sql.SparkSession;

import gov.va.sparkcql.Configuration;
import gov.va.sparkcql.adapter.model.ModelAdapter;
import gov.va.sparkcql.adapter.model.ModelAdapterFactory;

public class FhirModelAdapterFactory implements ModelAdapterFactory {

    @Override
    public ModelAdapter create(Configuration configuration, SparkSession spark) {
        return new FhirModelAdapter();
    }
}